editor/plugins/gradient_editor_plugin.h

```diff
@@ -33,25 +33,106 @@
 
 #include "editor/editor_inspector.h"
 #include "editor/editor_plugin.h"
-#include "gradient_editor.h"
+#include "scene/resources/gradient_texture.h"
 
-class GradientReverseButton : public BaseButton {
-	GDCLASS(GradientReverseButton, BaseButton);
+class EditorSpinSlider;
+class ColorPicker;
+class PopupPanel;
+class GradientTexture1D;
 
-	int margin = 2;
+class GradientEdit : public Control {
+	GDCLASS(GradientEdit, Control);
 
+	Ref<Gradient> gradient;
+	Ref<GradientTexture1D> preview_texture;
+
+	PopupPanel *popup = nullptr;
+	ColorPicker *picker = nullptr;
+
+	bool snap_enabled = false;
+	int snap_count = 10;
+
+	enum GrabMode {
+		GRAB_NONE,
+		GRAB_ADD,
+		GRAB_MOVE
+	};
+
+	GrabMode grabbing = GRAB_NONE;
+	float pre_grab_offset;
+	int pre_grab_index;
+	int selected_index = -1;
+	int hovered_index = -1;
+
+	// Make sure to use the scaled values below.
+	const int BASE_SPACING = 4;
+	const int BASE_HANDLE_WIDTH = 8;
```

Comment:
Should these be static?

Reply:
Not sure, I followed the old conventions in gradient_editor.cpp

Reply:
I think you can make them static, since there is another constant in the next class that is. I don't see why we'd want to have a non-static const here.